src/App.js

```diff
@@ -1,97 +1,91 @@
+    setPage(PAGE_HOME)
   }
 
-  async fetchBreweryData(searchTerm, searchByParam) {
-    try {
+  useEffect(() => {
+    async function fetchBreweryData() {
       const url = `https://api.openbrewerydb.org/breweries?by_${searchByParam}=${searchTerm}`
       const data = await fetch(url)
       const jsonData = await data.json()
-      const cleanData = this.filterResults(jsonData)
-      this.setState({
-        breweries: cleanData
-      })
-    } catch (error) {
-      console.log(error)
+      const cleanData = filterResults(jsonData)
+      setBreweries(cleanData)
     }
-  }
+    fetchBreweryData()
+  })
 
   // searchTerm is the value from the input/search bar
-  handleSearch = searchTerm => {
-    this.setState({
-      searchTerm,
-      page: PAGE_RESULTS
-    })
+  const handleSearch = searchTerm => {
+    setSearchTerm(searchTerm)
+    setPage(PAGE_RESULTS)
   }
 
-  searchBy = e => {
-    this.setState({
-      searchByParam: e
-    })
+  const searchBy = e => {
+    setSearchByParam(e)
   }
 
-  componentDidUpdate(previousProps, previousState) {
-    const { searchTerm, searchByParam } = this.state
-    const searchChanged = searchTerm !== previousState.searchTerm
-    if (searchChanged) {
-      this.fetchBreweryData(searchTerm, searchByParam)
-    }
-  }
+  // useEffect((previousProps, previousState) => {
+  //   const searchChanged = searchTerm !== previousState.searchTerm
+  //   if(searchChanged) {
+  //     this.fetchBreweryData(searchTerm, searchByParam)
+  //   }
+  // })
+
+
+//-------Original componentDidUpdate is below, unsuccessful attempt at hooks equivalent is above
+
+  // componentDidUpdate(previousProps, previousState) {
+  //   const { searchTerm, searchByParam } = this.state
+  //   const searchChanged = searchTerm !== previousState.searchTerm
+  //   if (searchChanged) {
+  //     this.fetchBreweryData(searchTerm, searchByParam)
+  //   }
+  // }
 
   // Exclude breweries that contains "Brewery In Planning"
-  filterResults = data =>
+  const filterResults = data =>
     data.filter(
       item =>
         !item.name.toLowerCase().includes('brewery in planning')
     )
 
-  whatToDisplay = page => {
-    const { breweries, searchTerm, searchByParam } = this.state
+  const whatToDisplay = page => {
     if (page === PAGE_RESULTS) {
       return (
         <BreweryList
           breweries={breweries}
-          backToSearch={this.backToSearch}
+          backToSearch={backToSearch}
           searchTerm={searchTerm}
           searchParam={searchByParam}
         />
       )
     }
     return (
       <BrewerySearch
-        handleSearch={this.handleSearch}
-        searchBy={this.searchBy}
+        handleSearch={handleSearch}
+        searchBy={searchBy}
       />
     )
   }
-
-  render() {
-    const { page } = this.state
     return (
       <Suspense fallback={<div>Loading...</div>}>
-        {this.whatToDisplay(page)}
+        {whatToDisplay(page)}
       </Suspense>
     )
-  }
 }
```

Comment:
Can you please change the whatToDisplay function name to something more meaningful? Also it doesn't need the page parameter/argument as the page state is available throughout App.

Reply:
This name is better, but the naming is too specific. This function can render either the BreweryList or the BrewerySearch component. Perhaps change it to something more generic like `displayBreweryComponent` or something in that sense. 

> Also it doesn't need the page parameter/argument as the page state is available throughout App.

`displayBreweryList(page)` still has the page argument.